### ISSUE TITLE
Curling published ports in docker swarm hangs

### ISSUE THREAD

**Issue by Nino-K** [rancher-sandbox/rancher-desktop#3221]
### Actual Behavior

Curling the docker swarm service with publish port hangs.

### Steps to Reproduce

```
# docker swarm init --advertise-addr 192.168.205.2

# docker network create --driver overlay --subnet 10.1.0.0/24 --gateway 10.1.0.1 svc_default

# cat docker-compose.yaml
version: "3.8"
services:
  swarm-test:
    image: nginx:latest
    deploy:
      replicas: 2
    ports:
      - mode: ingress
        protocol: tcp
        published: 8080
        target: 80
    networks:
      - svc_default

networks:
  svc_default:
    external: true

# docker stack deploy test -c docker-compose.yaml

# docker service ls
 
# curl -vvv http://localhost:8080/
```

### Result

Hangs when curling.

### Expected Behavior

For curl to suceed. succeed

### Additional Information

Although the issue was originally reported on 1.6 and M1 mac, I can confirm that this behavior is the same on 1.5.1 on a `x86_64` mac so it is not a regression and it is not specific to M1 CPU. 

Below is my observation, it seems as though the application listens on both IPV4 and IPV6, you can confirm this by curling the application from within the VM:

```
curl 127.0.0.1:8080
```
However, curling with IPV6:

```
curl -g -6 'http://localhost:8080'
```
Hangs, this could be due to this open issue: https://github.com/moby/moby/issues/24379

In additon, looking at the tcpdump within the VM I can see the curl request that is destined for port 8080 from the client (host in this case) is always forced to IPv6

```
20:02:55.367583 lo    In  IP6 ::1.54490 > ::1.8080: Flags [S], seq 3666321179, win 65476, options [mss 65476,sackOK,TS val 869371798 ecr 0,nop,wscale 7], length 0
20:02:55.367593 lo    In  IP6 ::1.8080 > ::1.54490: Flags [S.], seq 3999623390, ack 3666321180, win 65464, options [mss 65476,sackOK,TS val 869371798 ecr 869371798,nop,wscale 7], length 0
20:02:55.367602 lo    In  IP6 ::1.54490 > ::1.8080: Flags [.], ack 1, win 512, options [nop,nop,TS val 869371798 ecr 869371798], length 0
20:02:55.368230 lo    In  IP6 ::1.54490 > ::1.8080: Flags [P.], seq 1:79, ack 1, win 512, options [nop,nop,TS val 869371798 ecr 869371798], length 78: HTTP: GET / HTTP/1.1
20:02:55.368235 lo    In  IP6 ::1.8080 > ::1.54490: Flags [.], ack 79, win 511, options [nop,nop,TS val 869371798 ecr 869371798], length 0
20:02:59.929836 lo    In  IP6 ::1.54490 > ::1.8080: Flags [F.], seq 79, ack 1, win 512, options [nop,nop,TS val 869376360 ecr 869371798], length 0
20:02:59.977814 lo    In  IP6 ::1.8080 > ::1.54490: Flags [.], ack 80, win 511, options [nop,nop,TS val 869376408 ecr 869376360], length 0

```

This could potentially be related to the open issue mentioned above or perhaps misconfiguration in Lima’s underlying switch that always forces IPV6 with the swarm’s overlay networks. 

 However, it is worth noting that when a published port container is deployed:

```
docker run -p 8081:80 -td nginx
```

Everything works as expected and looking at the tcpdump in the VM the requests are made over IPV4 as expected:

```
19:34:54.386979 lo    In  IP 127.0.0.1.35410 > 127.0.0.1.8081: Flags [S], seq 3673555498, win 65495, options [mss 65495,sackOK,TS val 1782487875 ecr 0,nop,wscale 7], length 0
19:34:54.387035 lo    In  IP 127.0.0.1.8081 > 127.0.0.1.35410: Flags [S.], seq 228590965, ack 3673555499, win 65483, options [mss 65495,sackOK,TS val 1782487875 ecr 1782487875,nop,wscale 7], length 0
19:34:54.387092 lo    In  IP 127.0.0.1.35410 > 127.0.0.1.8081: Flags [.], ack 1, win 512, options [nop,nop,TS val 1782487875 ecr 1782487875], length 0
19:34:54.387974 lo    In  IP 127.0.0.1.35410 > 127.0.0.1.8081: Flags [P.], seq 1:79, ack 1, win 512, options [nop,nop,TS val 1782487876 ecr 1782487875], length 78
19:34:54.388017 lo    In  IP 127.0.0.1.8081 > 127.0.0.1.35410: Flags [.], ack 79, win 511, options [nop,nop,TS val 1782487876 ecr 1782487876], length 0
19:34:54.388655 lo    In  IP 127.0.0.1.8081 > 127.0.0.1.35410: Flags [P.], seq 1:854, ack 79, win 512, options [nop,nop,TS val 1782487877 ecr 1782487876], length 853
19:34:54.388703 lo    In  IP 127.0.0.1.35410 > 127.0.0.1.8081: Flags [.], ack 854, win 506, options [nop,nop,TS val 1782487877 ecr 1782487877], length 0
19:34:54.389693 lo    In  IP 127.0.0.1.35410 > 127.0.0.1.8081: Flags [F.], seq 79, ack 854, win 512, options [nop,nop,TS val 1782487878 ecr 1782487877], length 0
19:34:54.390404 lo    In  IP 127.0.0.1.8081 > 127.0.0.1.35410: Flags [F.], seq 854, ack 80, win 512, options [nop,nop,TS val 1782487879 ecr 1782487878], length 0
19:34:54.390534 lo    In  IP 127.0.0.1.35410 > 127.0.0.1.8081: Flags [.], ack 855, win 512, options [nop,nop,TS val 1782487879 ecr 1782487879], length 0

```

To summarize this could be a combination of the open issue mentioned above in conjunction with the misconfiguration of lima’s underlying switch (vde_vmnet) or the misbehavior is completely driven by the open issue here: https://github.com/moby/moby/issues/24379

### Rancher Desktop Version

1.6

### Rancher Desktop K8s Version

N/A

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

macOS

### Operating System / Build Version

Monterey 12.4 (21F79)

### What CPU architecture are you using?

x64

**Comment by richarddavenport** [comment]
@Nino-K I can confirm that I am seeing this behavior as well. I'm on MacOS too. I can also reproduce it on Windows.

Has there been any progress made to fix this?

**Comment by abiosoft** [comment]
What I did notice is that **_unless_** you access via the advertised address, docker would block the response.

Accessing via `192.168.205.2:8080` should work.

**Comment by madrover** [comment]
Same here. 
I would expect to access the published ports to be accesible on `localhost` as well.

**Comment by jmoseman01** [comment]
has this issue been addressed in version 10?
